buildSrc/subprojects/packaging/src/main/kotlin/org/gradle/gradlebuild/packaging/ShadedJarPlugin.kt

```diff
@@ -122,15 +122,19 @@ open class ShadedJarPlugin : Plugin<Project> {
         val baseVersion: String by rootProject.extra
         val jar: Jar by tasks
 
-        return tasks.create<ShadedJar>("${project.name}ShadedJar") {
-            dependsOn(jar)
-            jarFile.set(layout.buildDirectory.file("shaded-jar/${base.archivesBaseName}-shaded-$baseVersion.jar"))
-            classTreesConfiguration.from(configurationToShade.artifactViewForType(classTreesType))
-            entryPointsConfiguration.from(configurationToShade.artifactViewForType(entryPointsType))
-            relocatedClassesConfiguration.from(configurationToShade.artifactViewForType(relocatedClassesType))
-            manifests.from(configurationToShade.artifactViewForType(manifestsType))
-            buildReceiptFile.set(shadedJarExtension.buildReceiptFile)
-        }
+        val buildReceiptFile = project.layout.fileProperty().apply { set(shadedJarExtension.buildReceiptFile) }
+        val jarFile = project.layout.fileProperty().apply { set(layout.buildDirectory.file("shaded-jar/${base.archivesBaseName}-shaded-$baseVersion.jar")) }
+        val task = tasks.create<ShadedJar>(
+            "${project.name}ShadedJar",
+            project.files(configurationToShade.artifactViewForType(relocatedClassesType)),
```

Comment:
Let's not use `project.files` here. I only created a `ConfigureableFileCollection` in the first place to be able to modify it. This is not necessary anymore.